csm/core/services/usl.py

```diff
@@ -149,273 +146,104 @@ async def _format_lyve_pilot_access_params(
-                await s3_client.delete_bucket(bucket.name)
-            # Remove IAM user's access key if it exists
-            if iam_client is not None and iam_user_access_key is not None:
-                await iam_client.delete_user_access_key(
-                    iam_user_access_key.access_key_id,
-                    user_name=iam_user.user_name)
-            # Remove IAM user if it exists
-            if iam_client is not None and iam_user is not None:
-                await iam_client.delete_user(iam_user.user_name)
-            # Remove S3 account if it exists
-            if s3_account_service is not None and s3_account is not None:
-                s3_credentials = S3Credentials(
-                    str(s3_account.get('account_name')),
-                    str(s3_account.get('access_key')),
-                    str(s3_account.get('secret_key')),
-                    '',
-                )
-                try:
-                    await s3_account_service.delete_account(
-                        s3_credentials, s3_credentials.user_id
-                    )
-                except (S3ServiceError, Exception) as e:
-                    reason = f'Failed to remove account {s3_credentials.user_id}'
-                    Log.error(f'{reason} --- {e}')
-
-        Log.debug('Create S3 account')
+        iam_user: Optional[Dict] = None
+        iam_user_access_key: Optional[Dict] = None
+        iam_user_access_key_id: Optional[str] = None
+        bucket: Optional[Dict] = None
+
         try:
+            Log.debug('Create S3 account')
             s3_account = await s3_account_service.create_account(
-                s3_account_name, s3_account_email, s3_account_password,
-            )
-            # FIXME `cast()` is unsafe; mypy could not derive type correctly
-            s3_account = cast(Dict, s3_account)
-            access_key = s3_account.get('access_key')
-            secret_key = s3_account.get('secret_key')
+                s3_account_name, s3_account_email, s3_account_password)
             assert s3_account is not None
-        # FIXME There is no way to extract meaningful error codes from ``CsmInternalError``.
-        #   For that reason, we raise internal server errors in all of those cases.
+            s3_credentials = S3Credentials(
+                str(s3_account.get('account_name')),
+                str(s3_account.get('access_key')),
+                str(s3_account.get('secret_key')),
+                '')
+            Log.debug('Create IAM user')
+            iam_user = await s3_iam_users_service.create_user(
+                s3_credentials, iam_user_name, iam_user_password)
+            assert iam_user is not None
+            iam_user_arn = str(iam_user.get('arn'))
+            Log.debug('Create IAM user access key')
+            iam_user_access_key = await s3_access_keys_service.create_access_key(
+                s3_credentials, user_name=iam_user_name)
```

Comment:
Can `create_access_key()` return `None`? If not, since `iam_user_access_key` is not consumed afterwards, I would remove the assertion below and its declaration on line 202.

Reply:
1. No `create_access_key()` doesn't return `None`.
2. Removed.